### PR TITLE
type hints: models/vae.py

### PR DESCRIPTION
**Pull request by shepherd1530**
I have added type hints for 5 and 6 from #287 

@pcuenca I'll be glad if you can check it out.

### CODE REVIEW

File: src/diffusers/models/vae.py [hunk 1]
```diff
@@ -293,7 +293,7 @@ def __init__(self, parameters, deterministic=False):
         if self.deterministic:
             self.var = self.std = torch.zeros_like(self.mean).to(device=self.parameters.device)
 
-    def sample(self, generator=None):
+    def sample(self, generator=None) -> torch.Tensor:
```
Comment (patrickvonplaten) [inline src/diffusers/models/vae.py:296]:
```suggestion
    def sample(self, generator: Optional[torch.Generator] = None) -> torch.Tensor:
```

File: src/diffusers/models/vae.py [hunk 2]
```diff
@@ -429,18 +429,18 @@ def __init__(
...
 
-    def encode(self, x):
+    def encode(self, x: torch.FloatTensor) -> DiagonalGaussianDistribution:
         h = self.encoder(x)
         moments = self.quant_conv(h)
         posterior = DiagonalGaussianDistribution(moments)
         return posterior
 
-    def decode(self, z):
+    def decode(self, z: torch.Tensor) -> torch.FloatTensor:
```
Comment (patrickvonplaten) [inline src/diffusers/models/vae.py:438]:
```suggestion
    def decode(self, z: torch.FloatTensor) -> torch.FloatTensor:
```

File: src/diffusers/models/vae.py [hunk 3]
```diff
@@ -429,18 +429,18 @@ def __init__(
...
         z = self.post_quant_conv(z)
         dec = self.decoder(z)
         return dec
 
-    def forward(self, sample, sample_posterior=False):
+    def forward(self, sample: torch.FloatTensor, sample_posterior=False):
```
Comment (patrickvonplaten) [inline src/diffusers/models/vae.py:443]:
```suggestion
    def forward(self, sample: torch.FloatTensor, sample_posterior: bool = False) -> torch.FloatTensor:
```

**Review by patrickvonplaten — COMMENTED** [review]
Thanks a lot @shepherd1530 - left a couple of suggestions :-) Let me know if you have any questions

**Comment by shepherd1530** [comment]
@patrickvonplaten Thank you for the review. I have made the changes based on your suggestions. Thanks

**Review by patrickvonplaten — APPROVED** [review]
Great - merging!

**Comment by anton-l** [comment]
@patrickvonplaten something else blocking the merge?

**Pull request merged**